pkg/virtctl/imageupload/imageupload.go

```diff
@@ -133,7 +134,8 @@ func NewImageUploadCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 	cmd.Flags().StringVar(&size, "size", "", "The size of the DataVolume to create (ex. 10Gi, 500Mi).")
 	cmd.Flags().StringVar(&storageClass, "storage-class", "", "The storage class for the PVC.")
 	cmd.Flags().StringVar(&accessMode, "access-mode", "ReadWriteOnce", "The access mode for the PVC.")
-	cmd.Flags().BoolVar(&blockVolume, "block-volume", false, "Create a PVC with VolumeMode=Block (default Filesystem).")
+	cmd.Flags().StringVar(&volumeMode, "volume-mode", "", "The volume mode for the PVC, either Block or Filesystem.")
+	cmd.Flags().BoolVar(&blockVolume, "block-volume", false, "DEPRECATED - Create a PVC with VolumeMode=Block (default Filesystem).")
```

Comment:
Instead of adding a DEPRECATED here, could you leave the flag as-is and simply add:
```
cmd.Flags().MarkDeprecated("block-volume", "please use --volume-mode instead")
```

This will still allow people to use the flag, but they get  a warning like this printed:

```
Flag --block-volume has been deprecated, please use --volume-mode instead
```

Further the deprecated value will be hidden from the help text, so that people can still continue using it, but don't start adopting it by accident by looking at the help string.